### ISSUE TITLE
Failed to reproduce results on Imagenet.

### ISSUE THREAD

**Issue by Leun9** [zhengli97/CTKD#7]
Thanks for your great work.

We try to reproduce reults of KD (without CTKD) using `run_imagenet_distill.sh` but t it seems that something is wrong.

```
python3 train_student.py --path-t ./save/models/ResNet34/resnet34.pth \
        --distill kd \
        --batch_size 256 --epochs 120 --dataset imagenet \
        --gpu_id 0,1,2,3,4,5,6,7 --num_workers 32 \
        --learning_rate 0.1 --lr_decay_epochs 30,60,90 --weight_decay 1e-4 \
        --model_s ResNet18 -r 1 -a 1 -b 0 --kd_T 4 \
        ...
```

| |  RN34 | RN18 | KD |
| ------------- | ------------- | ------------- | ------------- |
| Paper  | 73.96  | 70.26  | 70.83  |
| Ours |  73.79 | 70.34  | 70.27, 70.48 |

I would like to know whether the cause is experimental error or parameter error.

**Comment by zhengli97** [comment]
This code was originally used for training on CIAFR-100. For training on ImageNet, I forgot to add the code to average the results of eight GPUs. I have updated my code. Please refer to `helper/loops.py line 249-258`. Add these code to your codebase and try again.

**Comment by Leun9** [comment]
> We used only one GPU for training. So maybe the results are not affected by this issue?

**Comment by zhengli97** [comment]
I didn't try to use one GPU for training on ImageNet. I always use 8* RTX 2080Ti for ImageNet. I guess this may be the reason for the different accuracy?

**Comment by Leun9** [comment]
> I didn't try to use one GPU for training on ImageNet. I always use 8* RTX 2080Ti for ImageNet. I guess this may be the reason for the different accuracy?

I guess this is an important reason.
Thanks for your reply.

**Comment by Leun9** [comment]
> I didn't try to use one GPU for training on ImageNet. I always use 8* RTX 2080Ti for ImageNet. I guess this may be the reason for the different accuracy?

How long is the training?

**Comment by zhengli97** [comment]
It will takes about one day for 120 epoch training
